src/espIdf/documentation/getDocsVersion.ts

```diff
@@ -111,3 +113,36 @@ export async function readObjectFromUrlFile(objectUrl: string) {
     return jsonic(objectMatches[0]);
   }
 }
+
+/**
+ * Retrieves the URL for the specified documentation part based on the ESP-IDF version and workspace.
+ * @param documentationPart - The documentation part to retrieve the URL for.
+ * @param workspace - The workspace URI.
+ * @returns The URL for the ESP-IDF specified documentation part.
+ */
+export async function getDocsUrl(
+  documentationPart: string,
+  workspace: vscode.Uri
+) {
+  const espIdfPath = idfConf.readParameter(
+    "idf.espIdfPath",
+    workspace
+  ) as string;
+
+  const adapterTargetName =
+    (idfConf.readParameter("idf.adapterTargetName", workspace) as string) ||
+    "esp32";
+  const idfVersion = await getEspIdfFromCMake(espIdfPath);
+  const docVersions = await getDocsVersion();
+  let docVersion = docVersions.find((docVer) => docVer.name === idfVersion);
```

Comment:
Will suggest to add validation in case idfVersion doesn't exist.

Reply:
I've added validation for a better user experience. User is warned if idfVersion is not found, the link for the documentation will contain the latest version